visualization/app/codeCharta/util/fileHelper.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { ExportBlacklistType, ExportCCFile } from "../codeCharta.api.model"
+import { AttributeTypeValue, BlacklistType } from "../codeCharta.model"
+import { getCCFile } from "./fileHelper"
+import { TEST_FILE_CONTENT } from "./dataMocks"
+const clone = require("rfdc")()
+
+describe("FileHelper", () => {
+	let fileContent: ExportCCFile
+
+	beforeEach(() => {
+		fileContent = clone(TEST_FILE_CONTENT)
+	})
+
+	describe("getCCFile", () => {
+		it("should build a CCFile", () => {
+			const result = getCCFile("fileName", fileContent)
+
+			expect(result).toMatchSnapshot()
+		})
+
+		it("should convert old blacklist type", () => {
+			fileContent.blacklist = [{ path: "foo", type: ExportBlacklistType.hide }]
+
+			const result = getCCFile("fileName", fileContent)
+
+			expect(result.settings.fileSettings.blacklist).toEqual([{ path: "foo", type: BlacklistType.flatten }])
+		})
+
+		it("should ignore old attribute types", () => {
+			fileContent.attributeTypes = {
+				nodes: [{ mcc: AttributeTypeValue.absolute }],
+				edges: [{ pairingRate: AttributeTypeValue.relative }]
+			}
```

Comment:
What do you think about deprecating this and providing a warning that this is not supported in a future major version?

Reply:
Sounds like a plan. Unfortunately we're lacking documentation of our api-version changes and I tried to sum it up as best as I could. I thought that we'd convert the old attribute types into our current schema and I was suprised that we just drop them.

I'll improve the validation and implement a warning.

Reply:
Yes, transforming them would likely even be better. I just don't know how long the old attributeTypes style was used.

Reply:
The main problem was, that the old attribute types were never really set in the analysis. Also the interfaces in the analysis were different than the one in the visualization. It was a mess and we fixed it alltogether.

I know it's not a good practice, but I think that we should actually start fresh, since our userbase is quite small and I don't think that there are many `.cc.json` files out there with old attribute types. Even when, those can easily be fixed by hand.

It's up to you though. :)

Reply:
I'll give it another thought.